src/Controller.cpp

```diff
@@ -63,12 +66,32 @@ extern "C"
         return (void *) (long) geopm_run_imp((struct geopm_ctl_c *)args);
     }
 
-    int geopmctl_main(void)
+    int geopmctl_main(int argc, char **argv)
     {
         int err = 0;
+        bool do_ctl_local = geopm::environment().do_ctl_local();
         try {
-            geopm::Controller ctl;
-            err = geopm_ctl_run((struct geopm_ctl_c *)&ctl);
+#ifdef GEOPM_ENABLE_MPI
+            if (!do_ctl_local) {
+                err = PMPI_Init(&argc, &argv);
+                if (err) {
+                    int str_size = NAME_MAX;
+                    char error_str[NAME_MAX + 1] = {};
+                    PMPI_Error_string(err, error_str, &str_size);
+                    fprintf(stderr, "Error: %s\n", error_str);
+                    return err;
+                }
+            }
+#endif
+            {
+                std::unique_ptr<geopm::Controller> ctl = std::make_unique<geopm::Controller>();
+                err = geopm_ctl_run((struct geopm_ctl_c *)ctl.get());
```

Comment:
maybe should just be calling  ``ctl->run()`` no need for the C interface anymore.